include/config.h

```diff
@@ -130,6 +131,16 @@ namespace vc4c
          * Maximum number of rounds the register-checker tries to resolve conflicts
          */
         unsigned registerResolverMaxRounds = 6;
+
+        /*
+         * Depth of loops whose constants are moved to out side of it
+         *
+         * * If it has no value, this optimization will not performed.
+         * * If it has negative value, all constants in loops will be moved.
+         *
+         * NOTE: This optimization is not enabled by default because it is incomplete.
+         */
+        Optional<int> moveConstantsDepth = {};
```

Comment:
Don't use optional here. use a simple integer with a default value. Whether the optimization is enabled, is determined by an additional command line option.

Reply:
What do you mean by "disable [...] by default"?
In the current version, the optimization is enabled by optimization level 3. Since the default optimization level is 2, it is disabled by default.
If you want it to be enabled ONLY when explicitly set via a switch, you can simply remove it from the optimization level 3.